File: solutions/ocaml/rna-transcription/1/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `U | `G | `C | `A ]

let to_rna =
  List.map (fun d -> match d with `A -> `U | `C -> `G | `G -> `C | `T -> `A)

File: solutions/fsharp/rna-transcription/1/RnaTranscription.fs
module RnaTranscription

let fromDna c =
    match c with
    | 'G' -> 'C'
    | 'C' -> 'G'
    | 'T' -> 'A'
    | 'A' -> 'U'
    | _ -> failwith "Invalid character"

// SURELY there is a less disgusting way to get it back into a string???
let toRna (dna: string) : string =
    dna.ToCharArray()
    |> Array.map fromDna
    |> fun result -> new System.String(result)
